assets/blocks/course-list-block/index.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { registerBlockVariation } from '@wordpress/blocks';
+import { list } from '@wordpress/icons';
+
+export const registerCourseListBlock = () => {
+	const DEFAULT_ATTRIBUTES = {
+		className: 'course-list-block',
+		query: {
+			perPage: 3,
+			pages: 0,
+			offset: 0,
+			postType: 'course',
+			order: 'desc',
+			orderBy: 'date',
+			author: '',
+			search: '',
+			exclude: [],
+			sticky: '',
+			inherit: false,
+		},
+	};
+
+	registerBlockVariation( 'core/query', {
+		name: 'sensei-lms/course-list',
+		title: __( 'Course List', 'sensei-lms' ),
+		description: __( 'Show a list of courses.', 'sensei-lms' ),
+		icon: list,
+		category: 'sensei-lms',
+		keywords: [
+			__( 'Course', 'sensei-lms' ),
+			__( 'List', 'sensei-lms' ),
+			__( 'Courses', 'sensei-lms' ),
+		],
+		attributes: { ...DEFAULT_ATTRIBUTES },
+		isActive: ( blockAttributes, variationAttributes ) => {
```

Comment:
This part is actually redundant. It behaves the same if it's here or not

Reply:
Sorry I'm not sure I've understood it correctly, can you please specify which line you are referring to?

Reply:
So I refer to the fact that this pattern with post type is added to Query Loop block also, regardless of this part of the code

Reply:
Yeah, patterns and variations both take post types and their behavior is almost interchangeable till now. In our case, it serves us best if we have them in both places